File: content-solutions/stack/dead-end-parking.java
import java.util.*;

class Solution {
    public List<Integer> solution(String[] commands) {
        Deque<Integer> lane = new ArrayDeque<>();
        List<Integer> answer = new ArrayList<>();
        for (String command : commands) {
            String[] parts = command.split(" ");
            if (parts[0].equals("in")) lane.push(Integer.parseInt(parts[1]));
            else if (!lane.isEmpty()) answer.add(lane.pop());
        }
        while (!lane.isEmpty()) answer.add(lane.pop());
        return answer;
    }
}

File: content-solutions/stack/dead-end-parking.py
def solution(commands):
    lane = []
    answer = []
    for command in commands:
        parts = command.split()
        if parts[0] == "in":
            lane.append(int(parts[1]))
        elif lane:
            answer.append(lane.pop())
    while lane:
        answer.append(lane.pop())
    return answer
